src/middleware/nasc.ts

```diff
@@ -108,13 +115,56 @@ async function NASCMiddleware(request: express.Request, response: express.Respon
 		if (pid) {
 			const linkedPIDs: number[] = device.linked_pids;
 
+			// * If a user performs a system transfer from
+			// * a console to another using a Nintendo account
+			// * during the transfer and both consoles have
+			// * a Pretendo account, the devices will be swapped
+			// * since we check it using the LFCS.
+			// *
+			// * So, the linked PIDs won't have the user's PID
+			// * anymore.
 			if (!linkedPIDs.includes(pid)) {
-				response.status(200).send(nascError('102').toString());
-				return;
+				const session: mongoose.ClientSession = await databaseConnection().startSession();
+				await session.startTransaction();
+
+				device.linked_pids.push(pid);
+
+				await device.save({ session });
+
+				await session.commitTransaction();
```

Comment:
Sessions are only needed when working with multiple collections which depend on the results of previous operations. For example during registration we create both a new `NEXAccount` and `Device`. However if something fails when creating the `Device`, we do not want the `NEXAccount` to be saved, that way the client can start "fresh" again at a later time

Sessions solves this problem by allowing you to stage multiple collection operates before committing to them, and allowing you to abort all of them at once. If any operation fails or the session is aborted then no changes take place for any of the collections used in the session